client.go

```diff
@@ -163,14 +163,16 @@ func (c *client) Scan(s *hrpc.Scan) ([]*hrpc.Result, error) {
 		if err != nil {
 			return nil, err
 		}
+
 		scanres = res.(*pb.ScanResponse)
 		results = append(results, scanres.Results...)
 
 		// TODO: The more_results field of the ScanResponse object was always
 		// true, so we should figure out if there's a better way to know when
 		// to move on to the next region than making an extra request and
 		// seeing if there were no results
-		for len(scanres.Results) != 0 {
+		for len(scanres.Results) != 0 &&
+				(numberOfRows == 0 || uint32(len(results)) < numberOfRows) {
```

Comment:
> `numberOfRows == 0` means there's no limit

could you please provide a reference to this? I'm looking at asynchbase (since gohbase is modeled from it) and it looks like they don't even allow a value of 0: https://github.com/OpenTSDB/asynchbase/blob/master/src/Scanner.java#L460